x/qgb/abci.go

```diff
@@ -0,0 +1,46 @@
+package qgb
+
+import (
+	"github.com/celestiaorg/celestia-app/x/qgb/keeper"
+	"github.com/celestiaorg/celestia-app/x/qgb/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// EndBlocker is called at the end of every block
+func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
```

Comment:
So with every block, we will be updating the validator set if it is different ? Doesn't this happen only after certain period ? not on every block ?

Reply:
it's possible that the validator set and how much voting power each has could change rapidly, so we have to check every block to ensure that we don't end up in a situation where the contracts can no longer be updated because the validator set changed too rapidly

Reply:
aaah, gotta make some more resources on validators in tendermint. Thanks

Reply:
would recommend looking into the [validator set](https://github.com/celestiaorg/celestia-core/blob/e0e519802a1a01059ac003ea627f3619b5d76214/types/validator_set.go#L113-L139), which is updated during `EndBlock` by the staking module. also related: the specs in the sdk are really great, and we will eventually add similar things to the qgb module https://github.com/cosmos/cosmos-sdk/tree/master/x/staking/spec

Reply:
Thanks